src/python/dxpy/bindings/__init__.py

```diff
@@ -657,6 +657,33 @@ def _wait_on_close(self, timeout=3600*24*1, **kwargs):
             i += 1
             elapsed += wait
 
+    def _wait_until_parts_uploaded(self, timeout=60, **kwargs):
```

Comment:
Is 60 seconds long enough to not trigger false re-uploads?

Reply:
I agree. However I was never able to reproduce the issue where certain part is not closed or it is still "in progress of download". Everytime code goes to this function, these parts were already closed.

I was wondering whether we should just re-upload it again without even waiting, but it seems to me that it might be the case that the part would be finished in a while.

Reply:
I share @emiloslavsky's concern on this one.  Part sizes can get pretty big so I would suggest a significantly longer timeout coupled with perhaps a ceilinged exponential backoff.  My guess is it will be hard to get historical metrics at the part-granularity so we will have to do judgement calls here.  We want to guard against releasing this and having uploads fail _en masse_ for many users.

Reply:
(Actually, I can query to see if logs on our file closers can give an idea here)